src/errors.exceptions.md

```diff
@@ -71,10 +74,24 @@ raise (ref MyError)(msg: "description", data: value)
 
 ### Practical notes
 
-The use of exceptions in some modules has significantly contributed to resource leaks, deadlocks and other difficult bugs. The various exception handling proposals aim to alleviate some of the issues but have not found sufficient grounding in the Nim community to warrant the language changes necessary to proceed.
+The use of exceptions in Nim has significantly contributed to resource leaks, deadlocks and other difficult bugs. The various exception handling proposals aim to alleviate some of the issues but have not found sufficient grounding in the Nim community to warrant the language changes necessary to proceed.
 
 A notable exception to the guideline is `chronos` and `async`/`await` transformations that lack support for propagating checked exception information. Several bugs and implementation issues exist in the exception handling transformation employed by `async`.
 
+### `Defect`
+
+`Defect` does [not cause](https://github.com/nim-lang/Nim/issues/12862) a `raises` effect - code must be manually verified - common sources of Defect include:
+
+* Over/underflows in signed arithmetic
+* `[]` operator for indexing arrays/seqs/etc (but not tables!)
+* accidental/implicit conversions to `range` types
+
+### `CatchableError`
+
+Catching `CatchableError` implies that any errors are funnelled through the same exception handler. When called code starts raising new exceptions, it becomes difficult to find affected code - catching more specific errors avoids this maintenance problem.
```

Comment:
`Result` has the same problem when you are not using enum error codes and yet this remains the vastly more popular choice in the codebases. Writing code based on the binary distinction "error vs no-error" is reasonable approach in many situations and it can be applied with exceptions as well.

Reply:
When `stringly` results are used, this is visible in the API at each abstraction level (ie each API makes an individual decision about this) and statically known at compile time - when it's used with exceptions, because of the dynamic typing of exceptions, the new raised type gets hidden by inheritance.

Since this is a page about exceptions, it's appropriate to cover how to avoid this specific problem when using exceptions. The page about `Result` details how to handle the issue when using Result (ie use an enum instead).